torch/jit/annotations.py

```diff
@@ -305,6 +308,8 @@ def try_ann_to_type(ann, loc):
         msg = "Unsupported annotation {} could not be resolved because {} could not be resolved."
         assert valid_type, msg.format(repr(ann), repr(contained))
         return OptionalType(valid_type)
+    if is_union(ann):
```

Comment:
Just making sure I understand the plan, we want to allow users to directly annotate values with `Union` from day 1?

Reply:
What is the plan here? It is a bit dangerous to enable it from day 1 because of FC. production services may not be updated to parse Union yet. But our serialized model from newer PyTorch versions would contain them, thus causing production crashes.